portal-bridge/src/bridge/state.rs

```diff
@@ -134,27 +153,152 @@ impl StateBridge {
 
             // gossip block's new state transitions
             for node in walk_diff.nodes.keys() {
-                let permit = gossip_send_semaphore
-                    .clone()
-                    .acquire_owned()
-                    .await
-                    .expect("to be able to acquire semaphore");
                 let account_proof = walk_diff.get_proof(*node);
-                let content_key = create_content_key(&account_proof)?;
-                let content_value =
-                    create_content_value(block_tuple.header.header.hash(), &account_proof)?;
-                Self::spawn_serve_state_proof(
-                    self.portal_client.clone(),
-                    content_key,
-                    content_value,
-                    permit,
-                    self.metrics.clone(),
-                );
+
+                // gossip the account
+                self.gossip_account(&account_proof, block_tuple.header.header.hash())
+                    .await?;
+
+                // check if node is a leaf then check if the leaf is for a contract, if it is gossip
```

Comment:
nit: this comment seems out of place now. Maybe remove it and add appropriate comments where they are needed